### PR TITLE
sqlccl: test prepared statement in session migration test

### PR DESCRIPTION
**Pull request by rafiss**
There are other tests in pkg/sql/testdata/session_migration but the one in this PR is a full integration test.

informs https://github.com/cockroachlabs/support/issues/2510
Release note: None

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
:lgtm: Thanks! Any luck with UDTs on a reproduction?

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/108305#-:-NbGPYbs-DUXIQv7-llV:b-5skr75)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ecwall and @jaylim-crl)

**Comment by rafiss** [comment]
Yeah, UDTs worked in the simple test I made. I updated the PR now.

File: pkg/ccl/testccl/sqlccl/show_transfer_state_test.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/jackc/pgx/v4"
```
Comment (ecwall) [inline pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:23]: Can you use v5 or is v4 required for this test?

Comment (rafiss) [reply pkg/ccl/testccl/sqlccl/show_transfer_state_test.go]: done

**Comment by rafiss** [comment]
tftr

bors r+

**Pull request merged**